File: 867. Transpose Matrix/soln1.py
# Runtime
# 70ms
# Beats 78.34%of users with Python3

# Memory
# 17.11MB
# Beats 53.63%of users with Python3

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        
        arr = [[0] * len(matrix) for i in range(len(matrix[0]))]

        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                arr[j][i] = matrix[i][j]
        return arr

File: 739. Daily Temperatures/soln1.cpp
// Runtime
// 124ms
// Beats 85.62% of users with C++

// Memory
// 105.42MB
// Beats 25.93% of users with C++

// used help from jk_Kdcgc on this solution
#include <stack>

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> indices;
        int n = temperatures.size();
        vector<int> ans(n);

        indices.push(0);
    
        for (int i = 1; i < n; i++){
    
            while (!indices.empty() && temperatures[indices.top()] < temperatures[i]){
                ans[indices.top()] = i - indices.top();
                indices.pop();
            }
            indices.push(i);
        }
        for (int i = 0; i < indices.size(); i++){
            ans[indices.top()] = 0;
            indices.pop();
        }

        return ans;
    }
};
